### ISSUE TITLE
Abstract ObjectType is allowed in Schema

### ISSUE THREAD

**Issue by travigd** [MichalLytek/type-graphql#564]
I have code that looks approximately like this.

```ts
@InterfaceType("Animal")
abstract class AnimalGQLInterface {
    @Field(() => String)
    abstract name: Promise<string>;
}

@ObjectType("AnimalBase", {
  isAbstract: true,
  implements: [AnimalGQLInterface],
})
abstract class Animal implements AnimalGQLInterface {
  abstract readonly id: string;

  async get name() {
    return nameService.getAnimalName(this.id);
  }
}

@ObjectType()
class Cat extends Animal {
  /* ... */
}

@ObjectType()
class Dog extends Animal {
  /* ... */
}
```

In other places, I accidentally wrote this:
```ts
// Incorrect: uses Animal instead of AnimalGQLInterface for decorator
@Field(() => [Animal])
pets: Animal[];
```

The correct code there was
```ts
@Field(() => [AnimalGQLInterface])
pets: Animal[];
```

Despite the fact that `isAbstract: true` was set on `Animal`, the resulting GraphQL schema includes `AnimalBase` as a type.

### What should it do?
It should throw an error that abstract type made it into the GraphQL schema.

**Comment by MichalLytek** [comment]
> Despite the fact that isAbstract: true was set on Animal, the resulting GraphQL schema includes AnimalBase as a type

`isAbstract` is only used to prevent auto-registering in schema. When you use it directly, it assumes you know what you're doing 😄 

> It should throw an error that abstract type made it into the GraphQL schema.

I think you're right 😉 Added to #12 🔒

**Comment by travigd** [comment]
My ultimate solution to this was to just remove the `@ObjectType` decorator on the base class (the main benefit of it was just the automatic `implements` which I can just manually add).

One thing that I think I noticed was that the `name` property on fields didn't work as expected (e.g., the TS property name was `getAge` and my decorator had `name: "age"`, and whenever I queried age, GraphQL complained that age was null since it was trying to access the property as `age` instead of `getAge`).

Is that something that's a known issue? It was also resolved when I stopped adding the `@ObjectType` decorator to the base class.
